test/libsolidity/semanticTests/functionTypes/selector_assignment_expression.sol

```diff
@@ -0,0 +1,14 @@
+contract D {
+    function f() external {}
+}
+
+contract C {
+    bool public z;
+    function h() public {
+        ((z = true) ? D.f : D.f).selector;
```

Comment:
```suggestion
        ((z = true) ? this.f : this.f).selector;
```

This should keep the example compilable even after we merge #14367.

Reply:
But contract `C` is not inheriting `D` in this case. But if we make it `contract C is D` then yes, it will also trigger the bug using `this.f`.

Reply:
Why is inheritance important? The bug can be triggered without inheritance. I mean, I tried a case like this locally, on develop and it fails via IR, which means it reproduces the bug.

Reply:
```solidity
contract C {
    bool public z;

    function f() external {}

    function h() public {
        ((z = true) ? this.f : this.f).selector;
    }
}
// ----
// h()
// z() -> true
```

Reply:
But I mean to use `this` in this example. Because there is not function `f` defined in `C`. So I guess you mean to change the whole example to this:
```solidity
contract C {
    bool public z;
    function f() public {
        ((z = true) ? this.f : this.f).selector;
    }
}
```

Reply:
> Should I add both cases or just replace by the one that you mentioned?

You can add it anyway. It's always good to increase test coverage.

Reply:
But in that case I guess no need to add it as a separate test case. These are just minor variations of the same thing. I would not mind more cases but actually ones that are substantially different from the ones we have already.